lib/Alchemy/Phrasea/Databox/Record/LegacyRecordRepository.php

```diff
@@ -129,25 +129,72 @@ public function findChildren(array $storyIds, $user = null)
         $connection = $this->databox->get_connection();
 
         $selects = $this->getRecordSelects();
-        array_unshift($selects, 's.rid_parent as story_id');
 
-        $builder = $connection->createQueryBuilder();
-        $builder
-            ->select($selects)
-            ->from('regroup', 's')
-            ->innerJoin('s', 'record', 'r', 'r.record_id = s.rid_child')
-            ->where(
-                's.rid_parent IN (:storyIds)',
-                'r.parent_record_id = 0'
-            )
-            ->setParameter('storyIds', $storyIds, Connection::PARAM_INT_ARRAY)
-        ;
+        if($max_items){
+            array_unshift($selects, 'sr.rid_parent as story_id');
+
+            $subBuilder = $connection->createQueryBuilder();
+
+            $subBuilder
+                ->select('s.*,
+                    IF(@old_rid_parent != s.rid_parent, @cpt := 1, @cpt := @cpt+1) AS CPT')
+                ->addSelect("IF(@old_rid_parent != s.rid_parent, IF(@old_rid_parent:=s.rid_parent,'NEW PARENT',0), '----------') AS Y")
+                ->from('regroup', 's')
+                ->where('s.rid_parent IN (:storyIds)')
+                ->setParameter('storyIds', $storyIds, Connection::PARAM_INT_ARRAY)
+                ->orderBy('s.rid_parent, s.ord')
+            ;
+
+            $builder = $subBuilder->getConnection()->createQueryBuilder();
+
+            $builder->select($selects)
+                ->from(sprintf('( %s )', $subBuilder->getSQL()), 'sr')
+                ->innerJoin('sr', 'record', 'r', 'r.record_id = sr.rid_child')
+                ->where('sr.CPT BETWEEN :offset AND :maxresult')
+                ->andWhere('r.parent_record_id = 0')
+                ->setParameter('offset', $offset)
+                ->setParameter('maxresult', ($offset + $max_items -1))
+                ->orderBy('story_id, sr.CPT')
+            ;
+
```

Comment:
the query can be built with one builder since this sql runs ok :
```
SET @cpt=1;
SET @old_rid_parent=-1;
SELECT
    r.coll_id AS collection_id,
    r.record_id,
    r.credate AS created,
    r.uuid,
    r.moddate AS updated,
    r.parent_record_id AS isStory,
    r.type,
    r.originalname AS originalName,
    r.sha256,
    r.mime,
    LPAD(BIN(r.status), 32, '0') as status
    , rid_parent as story_id
FROM
(
    SELECT
        r.*, g.rid_parent,
        IF(@old_rid_parent != rid_parent, @cpt := 1, @cpt := @cpt+1) AS CPT
    FROM
        (regroup AS g INNER JOIN record AS r ON(r.record_id=g.rid_child))
    ORDER BY
        rid_parent,
        ord
)
AS
    r
WHERE
    CPT BETWEEN 1 AND 5
```